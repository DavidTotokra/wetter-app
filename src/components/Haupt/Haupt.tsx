import { Content } from "./styles"

import { WetterData } from '../../model/Wetter'

export interface HauptProps {
  wetterData: WetterData;
  loading: boolean;
  error: boolean;
}

export function Haupt(props :HauptProps) {

  const wetterData = props.wetterData;
  const loading = props.loading;
  const error = props.error ;

  return (
    <Content>
     {  <a data-testid="link" href="https:Wetter.thb" aria-disableds>
        Wetter App
      </a> }

      {!loading && !error && (
        <div className="wrapper">
          <h1>{Math.round(wetterData.current.temp)}Â°</h1>
          <div className="info">
            <div className="stats">
              <h3>{wetterData.place.name}</h3>
              <span>
                {new Date().toLocaleTimeString('de-DE', {
                  timeZone: wetterData.timezone,
                  weekday: "long",
                  month: "short",
                  day: "numeric",
                  hour: "2-digit",
                  minute: "2-digit",
                })}
              </span>
            </div>
            <div className="status">
              <img src={`http://openweathermap.org/img/wn/${wetterData.current.weather[0].icon}@2x.png`} alt="wetter icon" />
              <span>{wetterData.current.weather[0].description}</span>
            </div>
          </div>
        </div>
      )}
    </Content>
  );
}