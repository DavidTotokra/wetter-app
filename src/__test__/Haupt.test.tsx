


import { render} from '@testing-library/react';
import '@testing-library/jest-dom/extend-expect';
import { Content } from '../components/Haupt/styles';






function renderHauptComponent() {
    return render(<Content />)
  }



describe('Haupt Componet', () => {
    

 
  it('should show the right text', () => {
    const { getByTestId } = renderHauptComponent();

    expect(getByTestId).toMatchSnapshot()
    
   
  });
  it('Should render correctly', () => {
    const Haup = render(<Content />)
    expect(Haup.container).toMatchSnapshot()
  })
  


    

});
