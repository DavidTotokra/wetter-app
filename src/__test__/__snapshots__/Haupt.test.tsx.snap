// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Haupt Componet Should render correctly 1`] = `
<div>
  <div
    class="sc-bczRLJ NKmwg"
  />
</div>
`;

exports[`Haupt Componet should show the right text 1`] = `[Function]`;
